src/bundler/bundle_v2.zig

```diff
@@ -7546,6 +7548,90 @@ pub const LinkerContext = struct {
+                            const parent_source_index = other_source_index;
+
+                            if (parent_result.top_level_await_keyword.len > 0) {
+                                tla_pretty_path = c.parse_graph.input_files.items(.source)[other_source_index].path.pretty;
+                                notes.append(Logger.Data{
+                                    .text = std.fmt.allocPrint(c.allocator, "The top-level await in {s} is here:", .{tla_pretty_path}) catch bun.outOfMemory(),
+                                }) catch bun.outOfMemory();
+                                break;
+                            }
+
+                            if (!Index.isValid(Index.init(parent_result.tla_check.parent))) {
+                                notes.append(Logger.Data{
+                                    .text = "unexpected invalid index",
+                                }) catch bun.outOfMemory();
+                                break;
+                            }
+
+                            other_source_index = parent_result.tla_check.parent;
+
+                            notes.append(Logger.Data{
+                                .text = std.fmt.allocPrint(c.allocator, "The file {s} imports the file {s} here:", .{
+                                    c.parse_graph.input_files.items(.source)[parent_source_index].path.pretty,
+                                    c.parse_graph.input_files.items(.source)[other_source_index].path.pretty,
+                                }) catch bun.outOfMemory(),
+                            }) catch bun.outOfMemory();
+                        }
+
+                        const imported_pretty_path = c.parse_graph.input_files.items(.source)[source_index].path.pretty;
+                        const text: string = if (strings.eql(imported_pretty_path, tla_pretty_path))
+                            std.fmt.allocPrint(c.allocator, "This require call is not allowed because the imported file \"{s}\" contains a top-level await", .{imported_pretty_path}) catch bun.outOfMemory()
+                        else
+                            std.fmt.allocPrint(c.allocator, "This require call is not allowed because the transitive dependency \"{s}\" contains a top-level await", .{tla_pretty_path}) catch bun.outOfMemory();
+
+                        const source: Logger.Source = c.parse_graph.input_files.items(.source)[source_index];
+                        c.log.addRangeErrorWithNotes(&source, record.range, text, notes.items) catch bun.outOfMemory();
+                    }
+                }
+            }
+
+            // Make sure that if we wrap this module in a closure, the closure is also
+            // async. This happens when you call "import()" on this module and code
+            // splitting is off.
+            if (Index.isValid(Index.init(result_tla_check.parent))) {
+                c.graph.meta.items(.flags)[source_index].is_async_or_has_async_dependency = true;
```

Comment:
we don't need to call .items here

we can hoist it out to the parent

Reply:
done https://github.com/oven-sh/bun/pull/14218/commits/6928acbb340241790c4d238d376129b8cc3939fa

Reply:
.items continues to be called